test/parallel/test-http-client-reject-unexpected-agent.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+require('../common');
+var assert = require('assert');
+var http = require('http');
+
+var options = {
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
+  agent: true,
+};
+
+try {
+  http.request(options);
+} catch (err) {
```

Comment:
Please use assert.throws() instead of try/catch.  This should fail but won't when http.request doesn't throw an exception.